tests/end-to-end/external-solution-integration/internal/scenario/connectivity_adapter/connectivity_adapter_e2e.go

```diff
@@ -0,0 +1,76 @@
+package connectivity_adapter
+
+import (
+	"fmt"
+
+	connectionTokenHandlerClient "github.com/kyma-project/kyma/components/connection-token-handler/pkg/client/clientset/versioned"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/internal"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/helpers"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testkit"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testsuite"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	payload    = "superduperpayload"
+	lambdaPort = 8080
+)
+
+// Steps return scenario steps
+func (s *CompassE2ELegacy) Steps(config *rest.Config) ([]step.Step, error) {
+	state, err := s.NewState()
+	if err != nil {
+		return nil, err
+	}
+
+	clients := testkit.InitKymaClients(config, s.testID)
+	compassClients := testkit.InitCompassClients(clients, state, s.domain, s.skipSSLVerify)
+
+	connectionTokenHandlerClientset := connectionTokenHandlerClient.NewForConfigOrDie(config)
+	appConnector := testkit.NewConnectorClient(
+		s.testID,
+		connectionTokenHandlerClientset.ApplicationconnectorV1alpha1().TokenRequests(s.testID),
+		internal.NewHTTPClient(s.skipSSLVerify),
+		log.New(),
+	)
+	testService := testkit.NewTestService(
+		internal.NewHTTPClient(s.skipSSLVerify),
+		clients.CoreClientset.AppsV1().Deployments(s.testID),
+		clients.CoreClientset.CoreV1().Services(s.testID),
+		clients.GatewayClientset.GatewayV1alpha2().Apis(s.testID),
+		s.domain,
+		s.testID,
+	)
+
+	lambdaEndpoint := helpers.LambdaInClusterEndpoint(s.testID, s.testID, lambdaPort)
+	apiServiceInstanceName := fmt.Sprintf("%s-api", s.testID)
+	eventServiceInstanceName := fmt.Sprintf("%s-event", s.testID)
+
+	return []step.Step{
+		step.Parallel(
+			testsuite.NewCreateNamespace(s.testID, clients.CoreClientset.CoreV1().Namespaces()),
+			testsuite.NewRegisterLegacyApplicationInCompass(s.testID, compassClients.DirectorClient, state),
+		),
+		step.Parallel(
+			testsuite.NewCreateMapping(s.testID, clients.AppBrokerClientset.ApplicationconnectorV1alpha1().ApplicationMappings(s.testID)),
+			testsuite.NewStartTestServer(testService),
+			testsuite.NewConnectApplicationUsingCompassLegacy(compassClients.ConnectorClient, appConnector, compassClients.DirectorClient, state),
+		),
+		testsuite.NewRegisterLegacyServiceInCompass(s.testID, testService.GetInClusterTestServiceURL(), compassClients.DirectorClient, testService, state),
```

Comment:
Consider changing the name, you are registering the application using legacy API, not registering a legacy application.